dagshub/data_engine/model/datasource.py

```diff
@@ -513,7 +513,34 @@ def func():
 
         return func()
 
-    def upload_metadata_from_dataframe(self, df: "pandas.DataFrame", path_column: Optional[Union[str, int]] = None):
+    def upload_metadata_from_file(self, file_path, path_column: Optional[Union[str, int]] = None, remote: bool = False):
+        """
+        Upload metadata from a file.
+
+        Args:
+            file_path: Path to the file with metadata. Allowed formats are CSV, Parquet, ZIP, GZ.
+            path_column: Column with the datapoints' paths. Can either be the name of the column, or its index.
+                If not specified, the first column is used.
+            remote: Whether to upload the metadata via remote or local upload. Default is False.
+                Remote upload means that the data engine API calls will be done from remote machine,
+                rather than from local machine which runs this client.
```

Comment:
[blocker]
This needs another name, remote is very unintuitive. Check the comments in the design doc
I don't have a good alternative suggestion though to be honest.

Reply:
Few suggestions, let me know what makes sense the most:
1. remote_upload
2. remote_execution
3. backgroud_processing

Reply:
I like Guy's suggestion of `ingest_on_server`, let me know what you think about it.
Maybe it can be shortened to just `serverside`.

Also alternative wording for the docstring, I'm rewording that because the client user doesn't really have a concept of "background processing", or what is a "remote machine":

```
ingest_on_server: Set to ``True`` to process the metadata asynchronously. The file will be sent to our server and ingested into the datasource there.
```

Reply:
`ingest_on_server` sounds pretty good. Agree about the "remote machine" and "background processing" terminology.